Switch to using WordPress Scripts and update packages

jom opened the pull request:
### Changes proposed in this Pull Request

* Updates `@wordpress` (and other) packages. With this, I tried to fix all the color issues that popped up with core's switch to CSS variables for WP Admin colors.
* Stop using Calypso build and switch to just use `@wordpress/scripts`. 

### Testing instructions

* TBC.

renatho commented:
Nice!!!
Just adding this comment here too for [documentation](https://github.com/Automattic/sensei/pull/4161#issuecomment-821373452) related to this PR. So after this PR is done, we could refactor all of our SVG icons to use [this](https://github.com/WordPress/gutenberg/tree/trunk/packages/scripts#using-svg) =)

jom commented:
Oof, I worked on this all day and couldn't get it working. Our dependencies are a bit of a mess. If anyone would like to make an attempt, please do 😃 

Some learnings:
- The closest I got was when I moved all `@wordpress` packages together based on their WordPress release. I then tried to trim as many indirect dependencies that we had added.
- Jest seems to have a few bugs when different versions intermix.
- The test breaks area a bit all over. Part of me feels like most/some of them will be cured with a Babel plugin that isn't present, but I'm not sure.
